cordapp-example/B9LAB-NOTES.md

```diff
@@ -0,0 +1,32 @@
+
+# B9lab Corda course notes
+
+## Module 3
+
+### Expand the example
+
+> To find out
+> - How do you pass a known state from the vault when launching a flow?
+
+I can use simply tx id, and in a code pull the state like so:
+`val ourStateRef = StateRef(SecureHash.parse(txId), 0)`
+
+> - What notary do you choose?
+
```

Comment:
Generally you should specify which notary you're going to use when you create new states:
```java
String notaryName = "O=Notary,L=London,C=GB";
Party notary = getServiceHub().getNetworkMapCache().getNotary(CordaX500Name.parse(notaryName));
```

If you're consuming states (i.e. your transaction has inputs); you should extract the notary from the input state:
```java
StateAndRef<IOU> txInput;
Party notary = txInput.getState().getNotary();
```